pkg/metric_generator/generator.go

```diff
@@ -101,3 +101,26 @@ func FilterMetricFamilies(l allowDenyLister, families []FamilyGenerator) []Famil
 
 	return filtered
 }
+
+// FilterMetricFamiliesLabels takes a AllowLabels and a slice of metric
+// families and returns a filtered slice with metric families with their functions adapted to include only allowed labels for that metric Family.
+func FilterMetricFamiliesLabels(allowLabels AllowLabels, families []FamilyGenerator) []FamilyGenerator {
```

Comment:
It feels like you are doing what `ComposeMetricGenFuncs` is doing + filtering labels here. How about calling this function with the metric families composed by `ComposeMetricGenFuncs` instead?

Reply:
Hey sorry for the late response. The thing is we need to exclude certain labels and AFAIK we can only inspect metric labels by wrapping the generateFunc?

Reply:
Indeed, you are right.